src/rte/RTESolver.jl

```diff
@@ -13,28 +13,81 @@ using ..RTE
 using ..Optics
 using ..LookUpTables
 
-export solve_lw!, solve_sw!
+export solve_lw!, solve_sw!, apply_metric_scaling!
 
 include("longwave1scalar.jl")
 include("longwave2stream.jl")
 include("shortwave1scalar.jl")
 include("shortwave2stream.jl")
 
 """
-    solve_lw!((; context, flux, src, bcs, op)::NoScatLWRTE, as::GrayAtmosphericState)
+    apply_metric_scaling!(flux::FluxLW, metric_scaling)
 
-Non-scattering RTE solver for the longwave problem, using gray optics.
+Apply metric scaling factor for longwave radiative fluxes. This accounts for geometric expansion of 
+grid columns with increasing altitude when deep-atmosphere metric terms are used. Requires
+`flux_up`, `flux_dn` and `flux_net` to be available in `flux`. The metric scaling factor is 
+an optional argument that has a scalar value of `1` when shallow atmosphere approximations
+are used (in which case this function returns nothing).
```

Comment:
```suggestion
are used (in which case this function does nothing).
```
The first method of this function should also `return nothing`. The current version returns `flux.flux_net`, which seems somewhat arbitrary.